modules/light-clients/08-wasm/types/store_test.go

```diff
@@ -1,11 +1,321 @@
+	// calls suite.SetupWasmWithMockVM() and creates two clients with their respective stores
+	subjectStore, substituteStore := suite.GetSubjectAndSubstituteStore()
+
+	testCases := []struct {
+		name     string
+		key      []byte
+		prefix   []byte
+		expStore storetypes.KVStore
+		expPanic error
+	}{
+		{
+			"success: subject store Delete",
+			host.ClientStateKey(),
+			types.SubjectPrefix,
+			subjectStore,
+			nil,
+		},
+		{
+			"failure: cannot Delete on substitute store",
+			host.ClientStateKey(),
+			types.SubstitutePrefix,
+			nil,
+			errors.New("key must be prefixed with subject/"),
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			wrappedStore := types.NewUpdateProposalWrappedStore(subjectStore, substituteStore)
+
+			prefixedKey := tc.prefix
+			prefixedKey = append(prefixedKey, tc.key...)
+
+			if tc.expPanic == nil {
+				wrappedStore.Delete(prefixedKey)
+
+				suite.Require().False(tc.expStore.Has(tc.key))
+			} else {
+				suite.Require().PanicsWithError(tc.expPanic.Error(), func() { wrappedStore.Delete(prefixedKey) })
+			}
+		})
+	}
+}
+
+// TestUpdateProposalWrappedStoreIterators tests the Iterator/ReverseIterator methods of the updateProposalWrappedStore.
+func (suite *TypesTestSuite) TestUpdateProposalWrappedStoreIterators() {
+	// calls suite.SetupWasmWithMockVM() and creates two clients with their respective stores
+	subjectStore, substituteStore := suite.GetSubjectAndSubstituteStore()
+
+	testCases := []struct {
+		name        string
+		start       []byte
+		end         []byte
+		prefixStart []byte
+		prefixEnd   []byte
+		expPanic    error
+	}{
+		{
+			"success: subject store Iterate",
+			[]byte("start"),
+			[]byte("end"),
+			types.SubjectPrefix,
+			types.SubjectPrefix,
+			nil,
+		},
+		{
+			"success: substitute store Iterate",
+			[]byte("start"),
+			[]byte("end"),
+			types.SubstitutePrefix,
+			types.SubstitutePrefix,
+			nil,
+		},
+		{
+			"failure: key not prefixed",
+			[]byte("start"),
+			[]byte("end"),
+			invalidPrefix,
+			invalidPrefix,
+			errors.New("key must be prefixed with either subject/ or substitute/"),
+		},
+		{
+			"failure: start and end keys not prefixed with same prefix",
+			[]byte("start"),
+			[]byte("end"),
+			types.SubjectPrefix,
+			types.SubstitutePrefix,
+			errors.New("start and end keys must be prefixed with the same prefix"),
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			wrappedStore := types.NewUpdateProposalWrappedStore(subjectStore, substituteStore)
+
+			prefixedKeyStart := tc.prefixStart
+			prefixedKeyStart = append(prefixedKeyStart, tc.start...)
+			prefixedKeyEnd := tc.prefixEnd
+			prefixedKeyEnd = append(prefixedKeyEnd, tc.end...)
+
+			if tc.expPanic == nil {
+				suite.Require().NotNil(wrappedStore.Iterator(prefixedKeyStart, prefixedKeyEnd))
+				suite.Require().NotNil(wrappedStore.ReverseIterator(prefixedKeyStart, prefixedKeyEnd))
```

Comment:
can't actually test iterators for equality. would be open to other ideas if people this its worth it.